desktop/libs/indexer/src/indexer/api3.py

```diff
@@ -737,3 +727,18 @@ def save_pipeline(request):
   response['message'] = request.POST.get('editorMode') == 'true' and _('Query saved successfully') or _('Notebook saved successfully')
 
   return JsonResponse(response)
+
+
+def upload_local_file(request):
+
+  upload_file = request.FILES['inputfile']
+  fs = FileSystemStorage()
+  name = fs.save(upload_file.name, upload_file)
+  reader = csv.reader(decode_utf8(upload_file))
+
+  csv_data.clear()
+
+  for row in reader:
+    csv_data.append(row)
```

Comment:
nit: FYI: probably there is a way to add all rows from the reader in one insutruction

actually, I see, you use a global `csv_data`, can we avoid this in a follow-up and reuse the `file_url` instead? (and we can also improve this file with a random key too later)